src/stop.sh

```diff
@@ -161,13 +161,23 @@ if [ $stop_all -eq 1 ]; then
         do_killcephadm
     fi
 
+    # killing processes
     for p in $ceph_osd ceph-mon ceph-mds ceph-mgr radosgw lt-radosgw apache2 ganesha.nfsd ; do
-        for try in 0 1 1 1 1 ; do
-            if ! pkill -u $MYUID $p ; then
-                break
+        pkill -u $MYUID $p
+    done
+
+    # wait for processes to close
+    for try in 1 1 2 3 5 ; do
+        do_break=true
+        for p in $ceph_osd ceph-mon ceph-mds ceph-mgr radosgw lt-radosgw apache2 ganesha.nfsd ; do
+            if ! pkill -u $MYUID $p -sigcont; then
```

Comment:
It is definitely better!
I am amazed; how did you found out that -0 does nothing?

Reply:
according to KILL(2)

> If sig is 0, then no signal is sent, but existence and permission checks are still performed; this can be used to check for the existence of a process ID or process group ID that the caller is permitted to signal.